evals/record.py

```diff
@@ -96,6 +97,24 @@ def as_default_recorder(self, sample_id: str):
     def current_sample_id(self) -> Optional[str]:
         return self._sample_id.get()
 
+    def pause(self):
+        sample_id = self.current_sample_id()
+        with self._event_lock:
+            if sample_id not in self._paused_ids:
+                self._paused_ids.append(sample_id)
+
+    def unpause(self):
+        sample_id = self.current_sample_id()
+        with self._event_lock:
+            if sample_id in self._paused_ids:
+                self._paused_ids.remove(sample_id)
+
+    def is_paused(self, sample_id: str = None):
+        if sample_id is None:
+            sample_id = self.current_sample_id()
```

Comment:
This isn't needed anymore since we are always passing non-None sample_id

Reply:
i may expect people to call is_paused() in the future, so decided to leave it in for now. thank you!